src/lib.rs

```diff
@@ -73,8 +73,8 @@ impl<'a, 'b> plugin::Plugin<Request<'a, 'b>> for Raw {
 
         if need_read {
             let max_length = req.get::<persistent::Read<MaxBodyLength>>()
-                .ok().cloned().unwrap_or(DEFAULT_BODY_LIMIT);
-            let body = try!(read_body_as_utf8(req, max_length));
+                .ok().unwrap_or(std::sync::Arc::new(DEFAULT_BODY_LIMIT));
```

Comment:
Sure. In terms of action items, basically just do what the old code did (that `cloned` actually just copies the integer out of the `Arc`).

Understanding-wise, `Arc` is like `Box` in the sense that it is a heap allocated pointer, just unlike `Box` it also allocates space for some atomic refcounts. In this case we don't really want to allocate space for a whole new `Arc` just to copy out of it on line 77, so it's best to copy the data out of the `Arc` returned by `get` by using `cloned` *before* we `unwrap_or`.